airflow/serialization/serializers/datetime.py

```diff
@@ -62,12 +68,30 @@ def deserialize(classname: str, version: int, data: dict | str) -> datetime.date
 
     from pendulum import DateTime
     from pendulum.tz import timezone
+    from pytz.reference import USTimeZone
+
+    tz: USTimeZone | Timezone | None = None
+    if isinstance(data, dict) and TIMEZONE in data:
+        if version == 1:
+            # try to deserialize unsupported US timezones
+            mapping_us_timezones = {
+                "EDT": USTimeZone(-5, "Eastern", "EST", "EDT"),
```

Comment:
in pytz these are already defined as Eastern, Central, Mountain and Pacific. No need to redefine I guess?

Reply:
I've also think that some other timezones have "short name", not only in the US. For example in the UK use BST (British Summer Time)

```python
from pendulum import timezone, DateTime
DateTime(2023, 8, 1, tzinfo=timezone("Europe/London")).tzname()
Out[3]: 'BST'
```

Reply:
To add insult to injury, they can be ambiguous:

```
BST, Bangladesh Standard Time, +6
BST, Bougainville Standard Time, +11
BST, British Summer Time, +1
```

source: https://gist.githubusercontent.com/alyssaq/f1f0ec50e79f1c089554d0de855dd09c/raw/ca2f858a1149a20aa9c7507e826eae5636e7cecd/timezone-abbreviations.csv

Reply:
Suggestion:

```
tz_map = {
  "EDT": Timezone("America/New_York"), # EDT is deprecated 
  "CEST": Timezone("CET"), # will be adjusted for DST automatically
}
```

and maybe a configurable one, so when an exception turns up the user can specify it in the config what the mapping needs to be? 


(https://en.wikipedia.org/wiki/List_of_tz_database_time_zones#NEW_YORK)

Reply:
>  to be nice on people experiencing issues now

What about simplifying that by not being that nice? IMHO fixing the issue in the v2 is sufficient.

Reply:
I'm okay with that, and having this out asap, as we haven't received any other reports yet. We can always add an additional fix.